lsp/eslint.lua

```diff
@@ -0,0 +1,204 @@
+local util = require 'lspconfig.util'
+local lsp = vim.lsp
+
+-- @brief
+--
+-- https://github.com/hrsh7th/vscode-langservers-extracted
+--
+-- `vscode-eslint-language-server` is a linting engine for JavaScript / Typescript.
+-- It can be installed via `npm`:
+--
+-- ```sh
+-- npm i -g vscode-langservers-extracted
+-- ```
+--
+-- `vscode-eslint-language-server` provides an `EslintFixAll` command that can be used to format a document on save:
+-- ```lua
+-- vim.lsp.config('eslint', {
+--   --- ...
+--   on_attach = function(client, bufnr)
+--     vim.api.nvim_create_autocmd("BufWritePre", {
+--       buffer = bufnr,
+--       command = "EslintFixAll",
+--     })
+--   end,
+-- })
+-- ```
+--
+-- See [vscode-eslint](https://github.com/microsoft/vscode-eslint/blob/55871979d7af184bf09af491b6ea35ebd56822cf/server/src/eslintServer.ts#L216-L229) for configuration options.
+--
+-- Messages handled in lspconfig: `eslint/openDoc`, `eslint/confirmESLintExecution`, `eslint/probeFailed`, `eslint/noLibrary`
+--
+-- Additional messages you can handle: `eslint/noConfig`
+
+local function fix_all(opts)
+  opts = opts or {}
+
+  local bufnr = util.validate_bufnr(0)
```

Comment:
```suggestion
  validate('bufnr', bufnr, 'number')
```

Reply:
The current code uses `util.validate_bufnr` intentionally. This function acts as a compatibility layer, ensuring buffer number validation works correctly even on Neovim v0.10, which `nvim-lspconfig` still supports.

The proposed `vim.validate` function was introduced later and is not available or might behave differently in v0.10. Therefore, sticking with `util.validate_bufnr` is necessary for maintaining backward compatibility.

Reply:
vim.lsp.config was introduced in 0.11, meaning that this code should only run on 0.11, so it's a perfectly fine change

Reply:
Sometimes, the obvious is so clear that it becomes hard to see :sweat_smile:

My bad, and thank you